static/src/javascripts/projects/common/modules/experiments/tests/weekend-reading-promo.js

```diff
@@ -21,24 +27,34 @@ define([
         this.expiry = '2016-09-12';
         this.author = 'Kate Whalen';
         this.description = 'For just one pageview, show users a banner promoting the Weekend Reading email';
-        this.audience = 1;
+        this.audience = 0.5;
         this.audienceOffset = 0;
         this.successMeasure = 'Snap banner promotes the Weekend Reading email and leads to visitors signing up';
         this.audienceCriteria = 'All visitors who have not yet seen this snap banner';
         this.dataLinkNames = 'weekend-reading-snap';
         this.idealOutcome = 'Visitors click on the CTA and sign-up to the Weekend Reading email';
 
         this.canRun = function () {
-            return !config.page.isAdvertisementFeature && config.page.contentType === 'Article' && !hasSeenDigestSnap();
+            return !config.page.isAdvertisementFeature &&
+                config.page.contentType === 'Article' &&
+                !isCompetingWithAdblockMessage() &&
+                // we ran a similar Snap before, so do not show our new CTA to these users
+                !hasSeenMessage('habit-digest-message-07-16');
         };
 
-        function hasSeenDigestSnap() {
-            // we ran a similar Snap before, so do not show our new CTA to these users
+        // check to see if user has an ad-blocker enabled; we want the ad-block message to have priority
+        function isCompetingWithAdblockMessage() {
```

Comment:
This will exclude people who have closed the ad-block membership message. Can't we remove this condition? Aren't all the clashes handled by `message.show()` now?